docs/import/manage_disaster_flexible.js

```diff
@@ -380,63 +524,74 @@ async function initializeFlexibleDisaster(assetData) {
- * waits for the list of columns to populate each column select. This is the
- * simplest initializer, because geography has no special cascading effects. The
- * other initializers {@link initializePoverty} and {@link initializeBuildings}
- * do this as well, but have additional tasks.
- * @return {Promise<*>} Promise that completes when display is complete
+ * kicks off request for the list of columns to populate each column select.
+ * This is the simplest initializer, because geography has no special cascading
+ * effects. The other initializers {@link initializePoverty} and
+ * {@link initializeBuildings} do at least this much, but have additional tasks.
+ * @return {Promise<boolean>} Promise that completes when disaster listing
+ *     finished, false if disaster changed. All initialize* functions have the
+ *     same return value
  */
 async function initializeGeography() {
   createSelectAndColumns('geography').on('change', onGeographyChange);
   if (await getAssetsAndSetOptionsForSelect(
           componentsData.geography.path, false)) {
-    return setInitialColumnValues('geography');
+    setInitialColumnValues('geography');
+    return true;
+  } else {
+    return false;
   }
 }
 
+const POVERTY_BUILDINGS_TEXT = 'Column with building count: ';
 /**
  * Initializes poverty-related display. In addition to doing the analog of what
  * {@link initializeGeography} does, there are two complications: geography div
  * visibility is based on poverty asset having geometries, and we must
  * initialize a column select (for the building-count column) in a different
  * div from the standard poverty div.
- * @return {Promise<void>} Promise that completes when all display completed
+ * @return {Promise<boolean>} Promise that completes when disaster listing
+ *     finished, false if disaster changed. All initialize* functions have the
+ *     same return value
  */
 async function initializePoverty() {
   const select = createSelectAndColumns('poverty').on(
       'change', () => onPovertyChange(select.val()));
   // Take a guess at what geography visibility will be.
   setGeographyDivVisibility(!getStoredValueFromPath(POVERTY_HAS_GEOMETRY_PATH));
   // Create special column select for building-count column.
-  $('#buildings-poverty-select-span')
+  $('#buildings-source-poverty-div')
       .empty()
+      .append(POVERTY_BUILDINGS_TEXT)
       .append(createSelectWithSimpleWriteOnChange(POVERTY_BUILDINGS_PATH));
   if (!await getAssetsAndSetOptionsForSelect(componentsData.poverty.path)) {
     // Disaster changed: abort.
-    return;
+    return false;
   }
   // Set actual geography visibility.
   await showGeographyDivBasedOnPoverty(select.val());
-  const propertyNames = await setInitialColumnValues('poverty');
-  if (propertyNames) {
-    // Do special column setup, if disaster unchanged.
-    setOptionsForPovertyBuildings(propertyNames);
-  }
+  setInitialColumnValues('poverty').then((propertyNames) => {
+    if (propertyNames) {
```

Comment:
why inside a then block now?, and sorry where are we checking disaster hasn't changed?

Reply:
Rephrased a bit, and fixed called functions' docs, which were stale about the fact that disaster change also triggers null return.

It's in a then block now because we're no longer waiting for it to finish before returning. The initialize* functions don't wait for column population, just for the asset pickers' population.